### PR TITLE
GA-ISSUE 303-304: [Pelagus] - Standardize wallet connection to dapps & Emit event when pelagus is initialized

### PR DESCRIPTION
**Pull request by ArtemBurakov**
Let's put this pull request on hold for a while and do more testing

### CODE REVIEW

**Comment by ArtemBurakov** [comment]
Let's put this pull request on hold for a while and do more testing

File: src/window-provider.ts [hunk 1]
```diff
@@ -1,49 +1,48 @@
...
-      window.postMessage(data, window.location.origin),
-    addEventListener: (fn: WindowListener) =>
-      window.addEventListener("message", fn, false),
-    removeEventListener: (fn: WindowListener) =>
-      window.removeEventListener("message", fn, false),
-    origin: window.location.origin,
-  }),
+// For 100% certainty we could create an iframe here, store the references and then destroy the iframe.
+// https://speakerdeck.com/fransrosen/owasp-appseceu-2018-attacking-modern-web-technologies?slide=95
+Object.defineProperty(window, TALLY_OBJECT_PROPERTY, {
```
Comment (rileystephens28) [inline src/window-provider.ts]: Let's remove the tally provider

File: window-provider/index.ts [hunk 2]
```diff
@@ -1,91 +1,30 @@
...
-  isUnlocked: false,
-  initialized: false,
-  isPermanentlyDisconnected: false,
-}
-
-export default class TallyWindowProvider extends EventEmitter {
-  // TODO: This should come from the background with onConnect when any interaction is initiated by the dApp.
-  // onboard.js relies on this, or uses a deprecated api. It seemed to be a reasonable workaround for now.
+export default class PelagusWindowProvider extends EventEmitter {
   chainId = "0x1"
```
Comment (rileystephens28) [inline window-provider/index.ts]: Change `chainId` to `0x2328`

File: window-provider/index.ts [hunk 3]
```diff
@@ -1,91 +1,30 @@
...
-
   selectedAddress: string | undefined
-
   connected = false
-
   isPelagus = true
-
   isTally = true
```
Comment (rileystephens28) [inline window-provider/index.ts]: Remove `isTally` field

File: window-provider/index.ts [hunk 4]
```diff
@@ -98,9 +37,6 @@ export default class TallyWindowProvider extends EventEmitter {
       }
     }
   >()
-
-  _state?: typeof METAMASK_STATE_MOCK
-
   providerInfo = {
     label: "Pelagus",
     injectedNamespace: "tally",
```
Comment (rileystephens28) [inline window-provider/index.ts]: `injectedNamespace` should be `pelagus`

**Review by rileystephens28 — APPROVED** [review]

**Pull request merged**